source/Missions/Services/Missiles/Handlers/MissileHandler.cs

```diff
@@ -0,0 +1,117 @@
+﻿using Common;
+using Common.Messaging;
+using Common.Network;
+using LiteNetLib;
+using Missions.Services.Agents.Packets;
+using Missions.Services.Missiles.Message;
+using Missions.Services.Network;
+using System;
+using TaleWorlds.Library;
+using TaleWorlds.MountAndBlade;
+
+namespace Missions.Services.Missiles.Handlers
+{
+    /// <summary>
+    /// Handler for missiles within a battle
+    /// </summary>
+    public interface IMissileHandler : IHandler, IDisposable
+    {
+
+    }
+
+    /// <inheritdoc/>
+    public class MissileHandler : IMissileHandler
+    {
+        readonly INetworkMessageBroker networkMessageBroker;
+        readonly INetworkAgentRegistry networkAgentRegistry;
+        public MissileHandler(INetworkMessageBroker networkMessageBroker, INetworkAgentRegistry networkAgentRegistry)
+        {
+            this.networkMessageBroker = networkMessageBroker;
+            this.networkAgentRegistry = networkAgentRegistry;
+
+            networkMessageBroker.Subscribe<AgentShoot>(AgentShootSend);
+            networkMessageBroker.Subscribe<NetworkAgentShoot>(AgentShootRecieve);
+        }
+
+        ~MissileHandler()
+        {
+            Dispose();
+        }
+
+        public void Dispose()
+        {
+            networkMessageBroker.Unsubscribe<AgentShoot>(AgentShootSend);
+            networkMessageBroker.Unsubscribe<NetworkAgentShoot>(AgentShootRecieve);
+        }
+
+        private void AgentShootRecieve(MessagePayload<NetworkAgentShoot> payload)
+        {
+            networkAgentRegistry.TryGetGroupController(payload.Who as NetPeer, out AgentGroupController agentGroupController);
```

Comment:
Return if this fails
```suggestion
            if (networkAgentRegistry.TryGetGroupController(payload.Who as NetPeer, out AgentGroupController agentGroupController) == false) return;
```